modular_citadel/code/game/machinery/toylathe.dm

```diff
@@ -0,0 +1,389 @@
+			updateUsrDialog()
+
+		if(href_list["category"])
+			selected_category = href_list["category"]
+			updateUsrDialog()
+
+		if(href_list["make"])
+
+			/////////////////
+			//href protection
+			being_built = stored_research.isDesignResearchedID(href_list["make"])
+			if(!being_built)
+				return
+
+			var/multiplier = text2num(href_list["multiplier"])
+			var/is_stack = ispath(being_built.build_path, /obj/item/stack)
+
+			/////////////////
+
+			var/coeff = (is_stack ? 1 : prod_coeff) //stacks are unaffected by production coefficient
+			var/metal_cost = being_built.materials[MAT_METAL]
+			var/glass_cost = being_built.materials[MAT_GLASS]
+			var/plastic_cost = being_built.materials[MAT_PLASTIC]
+			var/power = max(2000, (metal_cost+glass_cost+plastic_cost)*multiplier/5)
+
+			GET_COMPONENT(materials, /datum/component/material_container)
+			if((materials.amount(MAT_METAL) >= metal_cost*multiplier*coeff) && (materials.amount(MAT_GLASS) >= glass_cost*multiplier*coeff) && (materials.amount(MAT_PLASTIC) >= plastic_cost*multiplier*coeff))
+				busy = TRUE
+				use_power(power)
+				icon_state = "autolathe_n"
+				var/time = is_stack ? 32 : 32*coeff*multiplier
+				addtimer(CALLBACK(src, .proc/make_item, power, metal_cost, glass_cost, plastic_cost, multiplier, coeff, is_stack), time)
+
+		if(href_list["search"])
+			matching_designs.Cut()
+
+			for(var/v in stored_research.researched_designs)
+				var/datum/design/D = stored_research.researched_designs[v]
+				if(findtext(D.name,href_list["to_search"]))
+					matching_designs.Add(D)
+			updateUsrDialog()
+	else
+		to_chat(usr, "<span class=\"alert\">The autoylathe is busy. Please wait for completion of previous operation.</span>")
+
+	updateUsrDialog()
+
+	return
+
+/obj/machinery/autoylathe/proc/make_item(power, metal_cost, glass_cost, plastic_cost, multiplier, coeff, is_stack)
+	GET_COMPONENT(materials, /datum/component/material_container)
+	var/atom/A = drop_location()
+	use_power(power)
+	var/list/materials_used = list(MAT_METAL=metal_cost*coeff*multiplier, MAT_GLASS=glass_cost*coeff*multiplier, MAT_PLASTIC=plastic_cost*coeff*multiplier)
+	materials.use_amount(materials_used)
+
+	if(is_stack)
+		var/obj/item/stack/N = new being_built.build_path(A, multiplier)
+		N.update_icon()
+		N.autoylathe_crafted(src)
+	else
+		for(var/i=1, i<=multiplier, i++)
+			var/obj/item/new_item = new being_built.build_path(A)
+			for(var/mat in materials_used)
+				new_item.materials[mat] = materials_used[mat] / multiplier
+			new_item.autoylathe_crafted(src)
+	icon_state = "autolathe"
+	busy = FALSE
+	updateDialog()
+
+/obj/machinery/autoylathe/RefreshParts()
+	var/T = 0
+	for(var/obj/item/stock_parts/matter_bin/MB in component_parts)
+		T += MB.rating*75000
+	GET_COMPONENT(materials, /datum/component/material_container)
+	materials.max_amount = T
+	T=1.2
+	for(var/obj/item/stock_parts/manipulator/M in component_parts)
+		T -= M.rating*0.2
+	prod_coeff = min(1,max(0,T)) // Coeff going 1 -> 0,8 -> 0,6 -> 0,4
```

Comment:
use CLAMP instead of a min and a max at the same time.

Reply:
nani? CLAMP?